packages/web-components/src/divider/divider.ts

```diff
@@ -0,0 +1,38 @@
+import { attr } from '@microsoft/fast-element';
+import { FASTDivider } from '@microsoft/fast-foundation';
+import { DividerAlignContent, DividerAppearance } from './divider.options.js';
+
+/**
+ * @class Divider component
+ *
+ * @remarks
+ * This class extends the FASTDivider. A divider groups sections of content to create visual rhythm and hierarchy. Use dividers along with spacing and headers to organize content in your layout.
+ */
+export class Divider extends FASTDivider {
+  /**
+   * @property alignContent
+   * @default center (DividerAlignContent.center)
+   * @remarks
+   * Determines the alignment of the content within the divider. Select from center, start, or end.
+   */
+  @attr({ attribute: 'align-content' })
+  public alignContent?: DividerAlignContent;
```

Comment:
@chrisdholt - there is no unified approach to declaring optional attributes. We mix all the following:
- `badge.appearance` assigns a default value:
https://github.com/microsoft/fluentui/blob/eead74fee07339f998615fe34d8f847d0f63af6e/packages/web-components/src/badge/badge.ts#L18
- `progress-bar.shape` is marked as mandatory even though it should not be required:
https://github.com/microsoft/fluentui/blob/79e23c30ececb898858f9c6d9d1656af38cc487e/packages/web-components/src/progress-bar/progress-bar.ts#L27
- `avatar.avatar-size` in another PR uses `Type | undefined`:
```ts
public size: AvatarSize | undefined;
```
- this PR uses optionals.

Can we agree on a single thing? FUIR9 uses optionals.

Reply:
Is there a good reason to NOT set a default value?
public appearance: BadgeAppearance = BadgeAppearance.filled;

When I review a spec, one thing I look for is the default state of a given component. When we build and style the component accordingly, the consumer gets the default look and behavior without any further customizations. I would think this is a good thing.

As for optionals, this is a good practice that I picked up while working with another team. If I don't have a default value, I would opt for this approach.

Reply:
> Is there a good reason to NOT set a default value?
public appearance: BadgeAppearance = BadgeAppearance.filled;

The reason is the [`reflect` mode of Fast `@attr`](https://www.fast.design/docs/fast-element/defining-elements/#customizing-attributes).
It is a bidirectional binding, which will reflect any value assignment in Javascript to DOM.
```html
<!-- this is what I author in HTML -->
<fluent-badge></fluent-badge>

<!-- ⬇️ this is what I get in DOM -->
<fluent-badge appearance="filled" color="brand"></fluent-badge>
```

I see 2 problems with this:
1. It makes DOM bigger, which might be a performance concern.
2. It makes DOM different to HTML, which makes debugging more difficult and confusing (_"Where does the `color="brand"` come from? I've never set it anywhere."_

When I experienced this for the first time, I proposed to use `fromView` mode for all the attributes but that brings another problem: we depend on the attribute to be present in DOM in our CSS selectors:
https://github.com/microsoft/fluentui/blob/eead74fee07339f998615fe34d8f847d0f63af6e/packages/web-components/src/styles/partials/badge.partials.ts#L199

So we might use default value in attributes in `fromView` mode but we still could not depend on them in styles which would make the whole thing even more confusing.

For that reason, me and @chrisdholt decided to not use default attribute values, document what the default is and author the CSS appropriately.

Reply:
@miroslavstastny @chrisdholt I just pushed changes for this. Please let me know if this is what you meant by not using default attribute values.

Reply:
LGTM! Attributes marked as optional, no default value assigned, but documented in the jsdoc.
Default styles (when the prop is not set) correctly 'render' the default appearance, they never test for the default attribute value (in this case there is no `[align-content: 'center']` selector.
👍